listenbrainz/webserver/static/js/src/RecentListens.tsx

```diff
@@ -447,12 +463,61 @@ export default class RecentListens extends React.Component<
     }
   };
 
+  onChangeDateTimePicker = async (newDateTimePickerValue: Date | Date[]) => {
+    if (!newDateTimePickerValue) {
+      return;
+    }
+    this.setState({
+      dateTimePickerValue: newDateTimePickerValue,
+      loading: true,
+      lastFetchedDirection: "newer",
+    });
+    const { oldestListenTs, user } = this.props;
+    let minJSTimestamp;
+    if (Array.isArray(newDateTimePickerValue)) {
+      // Range of dates
+      minJSTimestamp = newDateTimePickerValue[0].getTime();
+    } else {
+      minJSTimestamp = newDateTimePickerValue.getTime();
+    }
+
+    // Constrain to oldest listen TS for that user
+    const minTimestampInSeconds = Math.max(
+      // convert JS time (milliseconds) to seconds
+      Math.round(minJSTimestamp / 1000),
+      oldestListenTs
+    );
+
+    const newListens = await this.APIService.getListensForUser(
+      user.name,
+      minTimestampInSeconds
+    );
+    if (!newListens.length) {
+      // No more listens to fetch
+      this.setState({
+        loading: false,
+      });
+      return;
+    }
+    this.setState(
+      {
+        listens: newListens,
+        nextListenTs: newListens[newListens.length - 1].listened_at,
+        previousListenTs: newListens[0].listened_at,
+        lastFetchedDirection: "newer",
```

Comment:
It seems we are not using `lastFetchedDirection` anywhere now. If that's the case, let's remove it.